### ISSUE TITLE
Generate prompt masks from JSONL

### ISSUE THREAD

**Issue by harubaru** [coreweave/kubernetes-cloud#182]
Currently, the tokenizer accepts plaintext and when fed through the [GPT Finetuner](https://github.com/coreweave/kubernetes-cloud/tree/master/finetuner-workflow) it optimizes on the entire context. For use-cases where optimization should only be performed on a target response, it would be useful to format training data in JSONL which would consist of request/response pairs in order to mask the request when calculating loss during training.

```json
{"prompt":"Overjoyed with the new iPhone! ->", "completion":" positive"}
{"prompt":"@lakers disappoint for a third straight night ->", "completion":" negative"}
```

In the end, the tokenizer should output a `.tokens` and `.mask` file, where `.mask` contains the masking for the prompts in `.tokens`.
